### PR TITLE
Custom Op example using rocBLAS calls

### PR DESCRIPTION
**Pull request by umangyadav**
Depends on https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/pull/1183 and https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/pull/1194.

This PR only adds an example using rocBLAS Calls.

Files to look at are in examples/ folder. i.e. custom_op_miopen_kernel.cpp, CMakeLists.txt and README.md .

### CODE REVIEW

File: examples/migraphx/custom_op_rocblas_kernel/custom_op_rocblas_kernel.cpp [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        assert(inputs[0].lengths().size() == 1);
+        assert(inputs[0].lengths()[0] == 1);
+        assert(inputs[1].lengths().size() == 1);
+        return inputs.back();
+    }
+};
+
+int main(int argc, const char* argv[])
+{
+    // computes ReLU(neg(x) * scale)
```
[the remark this answers is not in the document]
Comment (umangyadav) [reply examples/migraphx/custom_op_rocblas_kernel/custom_op_rocblas_kernel.cpp:78]: strictly speaking, we don't need that instruction, but same as before , that it is just for the face that, part of the program can be built using regular MIGraphX op and other part using custom op. Program using custom op can be built same way as any other migraphx program.

[the remark this answers is not in the document]
Comment (umangyadav) [reply examples/migraphx/custom_op_rocblas_kernel/custom_op_rocblas_kernel.cpp:78]: Another reason is, i plan to modify these examples with `runs_on_offload_target`  method later.  which just mean whether custom_op runs on GPU directly without needing to copying of buffer to/from Host. 

For all these examples right now, `runs_on_offload_target` is implicitly set to true. 

But later i can demonstrate how `runs_on_offload_target` method works without modifying exmaple much.

**Review by kahmed10 — APPROVED** [review]

File: examples/migraphx/custom_op_rocblas_kernel/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+
+list (APPEND CMAKE_PREFIX_PATH /opt/rocm/rocblas /opt/rocm/hip /opt/rocm)
+find_package (migraphx REQUIRED)
+find_package (rocblas REQUIRED)
+
+message("source file: " ${EXAMPLE}.cpp " ---> bin: " ${EXAMPLE})
+add_executable(${EXAMPLE} ${EXAMPLE}.cpp)
+
+target_link_libraries(${EXAMPLE} migraphx::c roc::rocblas hip::device)
```
Comment (pfultz2) [inline examples/migraphx/custom_op_rocblas_kernel/CMakeLists.txt:38]: This shouldn't link `hip::device` since its host-only APIs.

Comment (umangyadav) [reply examples/migraphx/custom_op_rocblas_kernel/CMakeLists.txt]: Removed

**Review by pfultz2 — APPROVED** [review]

**Pull request merged**